lib/prerequisite.js

```diff
@@ -9,6 +9,60 @@ module.exports = (input, pkg, opts) => {
 
 	const tasks = [
 		{
+			title: 'Ping npm registry',
+			task: () => new Promise((resolve, reject) => {
+				// `npm ping` waits wayyyy too long
+				const timeout = setTimeout(() => {
+					reject(new Error('Connection to npm registry timed out'));
+				}, 6000);
```

Comment:
You could use https://github.com/sindresorhus/p-timeout here instead of wrapping all this in a promise constructor.